presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -41,10 +43,45 @@
 import static com.facebook.presto.sql.gen.ByteCodeUtils.ifWasNullPopAndGoto;
 import static com.facebook.presto.sql.gen.ByteCodeUtils.invoke;
 import static com.facebook.presto.sql.gen.ByteCodeUtils.loadConstant;
+import static com.google.common.base.Verify.verify;
 
 public class InCodeGenerator
         implements ByteCodeGenerator
 {
+    enum SwitchGenerationCase
+    {
+        DIRECT_SWITCH,
+        HASH_SWITCH,
+        SET_CONTAINS,
+    }
+
+    static SwitchGenerationCase checkSwitchGenerationCase(Type type, List<RowExpression> values)
+    {
+        if (values.size() <= 1000) {
+            boolean areSmallPrimitives = false;
+            if (type instanceof BigintType || type instanceof DateType) {
+                areSmallPrimitives = true;
+                for (RowExpression expression : values) {
+                    if (expression instanceof ConstantExpression) {
```

Comment:
what about non-ConstantExpressions? If it's not possible for there to be one here then add an assertion